src/routes.tsx

```diff
@@ -36,269 +38,331 @@ enum EmbarkStory {
+      exact: true,
+    },
   },
   {
-    path: localePathPattern + '/new-member/connect-payment/direct',
-    Component: ConnectPaymentsDirectEntry,
-    exact: true,
+    path: `${onboardingLocaleBaseRoute}/connect-payment/direct`,
+    serverRouteData: {
+      titleTextKey: 'ONBOARDING_CONNECT_DD_PAGE_TITLE',
+    },
+    clientRouteData: {
+      Component: ConnectPaymentsDirectEntry,
+      exact: true,
+    },
   },
   {
-    path: localePathPattern + '/new-member/connect-payment/success',
-    Component: TrustlySpinnerPage,
-    exact: true,
+    path: `${onboardingLocaleBaseRoute}/connect-payment/success`,
+    serverRouteData: {
+      titleTextKey: 'ONBOARDING_CONNECT_DD_PAGE_TITLE',
+    },
+    clientRouteData: {
+      Component: TrustlySpinnerPage,
+      exact: true,
+    },
   },
   {
-    path: localePathPattern + '/new-member/connect-payment/fail',
-    Component: TrustlyFailPage,
-    exact: true,
+    path: `${onboardingLocaleBaseRoute}/connect-payment/fail`,
+    serverRouteData: {
+      titleTextKey: 'ONBOARDING_CONNECT_DD_PAGE_TITLE',
+    },
+    clientRouteData: {
+      Component: TrustlyFailPage,
+      exact: true,
+    },
   },
   {
-    path: localePathPattern + '/new-member/connect-payment/retry',
-    Component: TrustlySpinnerPage,
-    exact: true,
+    path: `${onboardingLocaleBaseRoute}/connect-payment/retry`,
+    serverRouteData: {
+      titleTextKey: 'ONBOARDING_CONNECT_DD_PAGE_TITLE',
+    },
+    clientRouteData: {
+      Component: TrustlySpinnerPage,
+      exact: true,
+    },
   },
   {
-    path: localePathPattern + '/new-member/(offer|sign)',
-    Component: OfferNew,
-    exact: true,
+    path: `${onboardingLocaleBaseRoute}/(offer|sign)`,
+    serverRouteData: {
+      titleTextKey: 'OFFER_PAGE_TITLE',
+    },
+    clientRouteData: {
+      Component: OfferNew,
+      exact: true,
+    },
   },
   {
-    path: localePathPattern + '/new-member/sign/success',
-    Component: SignLoading,
-    exact: true,
+    path: `${onboardingLocaleBaseRoute}/sign/success`,
+    serverRouteData: {
+      titleTextKey: '',
+    },
+    clientRouteData: {
+      Component: SignLoading,
+      exact: true,
+    },
   },
   {
-    path: localePathPattern + '/new-member/sign/fail',
-    Component: SignLoading,
-    exact: true,
+    path: `${onboardingLocaleBaseRoute}/sign/fail`,
+    serverRouteData: {
+      titleTextKey: '',
+    },
+    clientRouteData: {
+      Component: SignLoading,
+      exact: true,
+    },
   },
   {
-    path: localePathPattern + '/new-member/(offer|sign)/:id',
-    Component: OfferPage,
-    exact: true,
+    path: `${onboardingLocaleBaseRoute}/(offer|sign)/:id`,
+    serverRouteData: {
+      titleTextKey: 'OFFER_PAGE_TITLE',
+    },
+    clientRouteData: {
+      Component: OfferPage,
+      exact: true,
+    },
   },
   {
-    path: localePathPattern + '/new-member/debugger',
-    Component: Debugger,
-    exact: true,
+    path: `${onboardingLocaleBaseRoute}/checkout/details`,
```

Comment:
These URLs will need to include quote cart ID :)

```suggestion
    path: `${onboardingLocaleBaseRoute}/offer/:cartId/checkout/details`,
```

Reply:
Ah, true! Because we don't want to store it in e.g. local storage?

Reply:
I think it makes for a more standard and cleaner solution :)

- We can build a link to the offer page in a super simple way
- We can server render the page if we want to
- No need to clean up stale state in local storage

Reply:
Made the routes like `${onboardingLocaleBaseRoute}/checkout/details/:id` since `${onboardingLocaleBaseRoute}/offer/:cartId/checkout/details` looked a bit strange to me when actually having a quote cart ID. Named the URL parameter `id` since that's what we call it for the `(offer|sign)` route, but I think your `cartId` is probably better.

Reply:
Could you please elaborate on why you think it "looked a bit strange to me when actually having a quote cart ID"?

Reply:
@mehdielket I went with this option since I think the first one looks cluttered with that long ID in the middle. Given that we’re reading from left to right it seems reasonable to me to have the information more relevant to the user as far left as possible, for the few users who would have a glance on the URL. Those users probably are more interested in seeing that this page is part of the **checkout** than reading some string of characters that makes no sense to them.

You could also argue that this follows the pattern of `/offer/:quoteCartId`, which I think is as consistent and make as much sense as having the checkout routes as sub-routes to the  `/offer/:quoteCartId` route.

See [this Notion page](https://www.notion.so/hedviginsurance/OTHER-NOTES-e4d18b73f46f4f188d1daad52cab12a9) for a comparison between the URL's (and the same text with my thoughts as the one posted above).